### PR TITLE
Product page

### PR DESCRIPTION
**Pull request by amartya-rs**
feat: Product listing page

## Files to check

- [src/components/product-card/ProductCard.jsx ] 
- [ src/components/sidebar/Sidebar.jsx] 
- [ src/pages/product-page/ProductPage.jsx]

### CODE REVIEW

File: src/components/sidebar/Sidebar.jsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        <aside className="px-4 py-1 my-2">
+            <div className="aside-heading">
+                <h5>Filters</h5>
+                <button className="button-link">Clear</button>
+            </div>
+            <hr />
+            <h6 className="my-1">Category</h6>
+            <ul className="side-nav font-medium pl-1">
+                <li>
+                    <input id="category-1" type="checkbox" />
```
Comment (rutu14) [inline src/components/sidebar/Sidebar.jsx:12]: Suggestion: 
Can have a checkbox component and reuse the same.
Same can been done for radio btn components.

File: src/components/product-card/ProductCard.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { CartIcon, HeartIcon } from "../../assets/icons";
+import { sold_out_img } from "../../assets/image";
+import "./product-card.css";
+
+const ProductCard = ({ card }) => {
+   //calculating the discount percent of products
+   const discount = () => {
```
Comment (rutu14) [inline src/components/product-card/ProductCard.jsx:7]: Suggestion: Can reduce the number of lines for this function.

**Review by rutu14 — APPROVED** [review]
Overall, the code is good to merge.

File: src/components/sidebar/Sidebar.jsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                    <label htmlFor="category-4">Longsleeve</label>
+                </li>
+            </ul>
+            <hr />
+            <h6 className="my-1">Rating</h6>
+            <ul className="side-nav font-medium pl-1">
+                <li>
+                    <input id="rating-4" name="rating" type="radio" />
+                    <label htmlFor="rating-4">4 stars and above</label>
+                </li>
```
Comment (zoef-yt) [inline src/components/sidebar/Sidebar.jsx:34]: repeating elements can be converted into components and reused using props.

File: src/components/sidebar/Sidebar.jsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                <li>
+                    <input id="category-2" type="checkbox" />
+                    <label htmlFor="category-2">Hoodie</label>
+                </li>
+                <li>
+                    <input id="category-3" type="checkbox" />
+                    <label htmlFor="category-3">Cap</label>
+                </li>
+                <li>
+                    <input id="category-4" type="checkbox" />
```
Comment (iRohitGaur) [inline src/components/sidebar/Sidebar.jsx:24]: suggestion: see if you need the id here

File: src/components/sidebar/Sidebar.jsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+                    <label htmlFor="category-4">Longsleeve</label>
+                </li>
+            </ul>
+            <hr />
+            <h6 className="my-1">Rating</h6>
+            <ul className="side-nav font-medium pl-1">
+                <li>
+                    <input id="rating-4" name="rating" type="radio" />
```
Comment (iRohitGaur) [inline src/components/sidebar/Sidebar.jsx:32]: suggestion: make a radio input component and reuse it

**Review by iRohitGaur — APPROVED** [review]
Nice work! Good to merge with changes 👍🏻

**Review by zoef-yt — APPROVED** [review]
The code looks good, Do the suggested changes. also, it would be better to have a preview easier to visualize or screenshots also work.

File: src/components/sidebar/Sidebar.jsx (hunk 3 above)
Comment (amartya-rs) [reply src/components/sidebar/Sidebar.jsx:34]: Will work on it. Thanks for pointing it out.

File: src/components/sidebar/Sidebar.jsx (hunk 4 above)
Comment (amartya-rs) [reply src/components/sidebar/Sidebar.jsx:24]: I am using id to bind the labels with the corresponding inputs

File: src/components/product-card/ProductCard.jsx (hunk 2 above)
Comment (amartya-rs) [reply src/components/product-card/ProductCard.jsx:7]: Thanks for your suggestion

**Pull request merged**